RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -0,0 +1,383 @@
+
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryParametrization/interface/CurvilinearTrajectoryError.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TrackingTools/TrajectoryState/interface/FreeTrajectoryState.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/DetLayers/interface/DetLayer.h"
+
+#include "RecoMuon/TrackingTools/interface/MuonServiceProxy.h"
+#include "RecoMuon/TrackingTools/interface/MuonPatternRecoDumper.h"
+
+using namespace std;
+using namespace edm;
+using namespace l1t;
+
+// constructors
+L2MuonSeedGeneratorFromL1T::L2MuonSeedGeneratorFromL1T(const edm::ParameterSet& iConfig) : 
+  theSource(iConfig.getParameter<InputTag>("InputObjects")),
+  theL1GMTReadoutCollection(iConfig.getParameter<InputTag>("GMTReadoutCollection")), // to be removed
+  thePropagatorName(iConfig.getParameter<string>("Propagator")),
+  theL1MinPt(iConfig.getParameter<double>("L1MinPt")),
+  theL1MaxEta(iConfig.getParameter<double>("L1MaxEta")),
+  theL1MinQuality(iConfig.getParameter<unsigned int>("L1MinQuality")),
+  useOfflineSeed(iConfig.getUntrackedParameter<bool>("UseOfflineSeed", false)),
+  useUnassociatedL1(iConfig.existsAs<bool>("UseUnassociatedL1") ? 
+		    iConfig.getParameter<bool>("UseUnassociatedL1") : true),
+  centralBxOnly_( iConfig.getParameter<bool>("CentralBxOnly") )
+  {
+
+  muCollToken_ = consumes<MuonBxCollection>(theSource);
+
+  if(useOfflineSeed) {
+    theOfflineSeedLabel = iConfig.getUntrackedParameter<InputTag>("OfflineSeedLabel");
+    offlineSeedToken_ = consumes<edm::View<TrajectorySeed> >(theOfflineSeedLabel);
+  }
+  
+  // service parameters
+  ParameterSet serviceParameters = iConfig.getParameter<ParameterSet>("ServiceParameters");
+  
+  // the services
+  theService = new MuonServiceProxy(serviceParameters);
+
+  // the estimator
+  theEstimator = new Chi2MeasurementEstimator(10000.);
+
+
+  produces<L2MuonTrajectorySeedCollection>(); 
+}
+
+// destructor
+L2MuonSeedGeneratorFromL1T::~L2MuonSeedGeneratorFromL1T(){
+  if (theService)   delete theService;
+  if (theEstimator) delete theEstimator;
+}
+
+void
+L2MuonSeedGeneratorFromL1T::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("GMTReadoutCollection", edm::InputTag("")); // to be removed
+  desc.add<edm::InputTag>("InputObjects",edm::InputTag("hltGmtStage2Digis"));
+  desc.add<string>("Propagator", "");
+  desc.add<double>("L1MinPt",-1.);
+  desc.add<double>("L1MaxEta",5.0);
+  desc.add<unsigned int>("L1MinQuality",0);   
+  desc.addUntracked<bool>("UseOfflineSeed",false);
+  desc.add<bool>("UseUnassociatedL1", true);
+  desc.add<bool>("CentralBxOnly", true);
```

Comment:
the existAs is not needed as the parameter is in the fillDescriptions.
Parameters added like this are fine...if the default is properly set!